addons/sales_team/views/crm_team_views.xml

```diff
@@ -196,5 +196,14 @@
             parent="sales_team.menu_sale_config"
             action="sales_team_config_action"/>
 
+        <record id="sales_team_config_mail_activity_type_action" model="ir.actions.act_window">
+            <field name="name">Activity Types</field>
+            <field name="res_model">mail.activity.type</field>
+            <field name="view_type">form</field>
+            <field name="view_mode">tree,form</field>
+            <field name="domain">['|', ('res_model_id','=', False), ('res_model_id','in',['crm.lead', 'res.partner', 'sale.order', 'product.template', 'product.product'])]</field>
```

Comment:
This one is a tricky one. Probably in crm you should set a domain in lead module on lead and partner; one in sales on sale order and product; and one in sale_crm that takes both of them. Do you think this could enforce modularity and still work as requested ?